core/commands/ping.go

```diff
@@ -0,0 +1,148 @@
+package commands
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"time"
+
+	cmds "github.com/jbenet/go-ipfs/commands"
+	peer "github.com/jbenet/go-ipfs/p2p/peer"
+	u "github.com/jbenet/go-ipfs/util"
+
+	context "github.com/jbenet/go-ipfs/Godeps/_workspace/src/code.google.com/p/go.net/context"
+)
+
+const kPingTimeout = 10 * time.Second
+
+type PingResult struct {
+	Success bool
+	Time    time.Duration
+	Text    string
+}
+
+var PingCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "send echo request packets to IPFS hosts",
+		Synopsis: `
+ipfs ping <peer.ID> - Send pings to a peer using the routing system to discover its address
```

Comment:
Oooh, I didnt allow specifying a multiaddr. Do you want that implemented? Or should I just reword?